src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -1364,8 +1375,371 @@ def get_unit_spike_train(
+    storage_options : dict | None, default: None
+        Additional kwargs (e.g. AWS credentials) passed to zarr.open. Only used with
+        "zarr" stream_mode.
+    use_pynwb : bool, default: False
+        If True, uses pynwb library to read the NWB file. Default False uses h5py/zarr
+        directly for better performance.
+
+    Returns
+    -------
+    recording : NwbTimeSeriesExtractor
+        A recording extractor containing the TimeSeries data.
+    """
+
+    installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+
+    def __init__(
+        self,
+        file_path: str | Path | None = None,
+        timeseries_path: str | None = None,
+        load_time_vector: bool = False,
+        samples_for_rate_estimation: int = 1_000,
+        stream_mode: Optional[Literal["fsspec", "remfile", "zarr"]] = None,
+        stream_cache_path: str | Path | None = None,
+        *,
+        file: BinaryIO | None = None,
+        cache: bool = False,
+        storage_options: dict | None = None,
+        use_pynwb: bool = False,
+    ):
+        if file_path is not None and file is not None:
+            raise ValueError("Provide either file_path or file, not both")
+        if file_path is None and file is None:
+            raise ValueError("Provide either file_path or file")
+
+        self.file_path = file_path
+        self.stream_mode = stream_mode
+        self.stream_cache_path = stream_cache_path
+        self.storage_options = storage_options
+        self.timeseries_path = timeseries_path
+
+        if self.stream_mode is None and file is None:
+            self.backend = _get_backend_from_local_file(file_path)
+        else:
+            self.backend = "zarr" if self.stream_mode == "zarr" else "hdf5"
+
+        if use_pynwb:
+            try:
+                import pynwb
+            except ImportError:
+                raise ImportError(self.installation_mesg)
+
+            channel_ids, sampling_frequency, dtype, segment_data, times_kwargs = self._fetch_recording_segment_info(
+                file, cache, load_time_vector, samples_for_rate_estimation
+            )
+        else:
+            channel_ids, sampling_frequency, dtype, segment_data, times_kwargs = (
+                self._fetch_recording_segment_info_backend(file, cache, load_time_vector, samples_for_rate_estimation)
+            )
+
+        BaseRecording.__init__(self, channel_ids=channel_ids, sampling_frequency=sampling_frequency, dtype=dtype)
+        recording_segment = NwbTimeSeriesSegment(
+            timeseries_data=segment_data,
+            times_kwargs=times_kwargs,
+        )
+        self.add_recording_segment(recording_segment)
+
+        if storage_options is not None and stream_mode == "zarr":
+            warnings.warn(
+                "The `storage_options` parameter will not be propagated to JSON or pickle files "
+                "for security reasons, so the extractor will not be JSON/pickle serializable."
```

Comment:
```suggestion
                "for security reasons, so this extractor will not be JSON/pickle serializable."
```
the is not wrong, but this emphasizes that it the current version of the extractor. Up to you if you want that emphasis.

Reply:
Makes sense. Thanks.